sdk/include/opentelemetry/sdk/metrics/instruments.h

```diff
@@ -19,6 +19,9 @@ enum class InstrumentType
   kCounter,
   kHistogram,
   kUpDownCounter,
+#if OPENTELEMETRY_ABI_VERSION_NO >= 2
+  kGauge,
+#endif
```

Comment:
The SDK can implement more features like kGauge even when these are not reachable from ABI v1, because there is no CreateIntGauge() virtual method in the API for ABI V1.

So, it should be just fine to always declare kGauge here, without if OPENTELEMETRY_ABI_VERSION_NO >= 2.

Reply:
Good point. Have removed the OPENTELEMETRY_ABI_VERSION_NO from SDK where possible.

Reply:
Should we append `kGuage` at the end of the enum so other values are still backward compatible?